packages/core/src/telemetry/loggers.ts

```diff
@@ -422,6 +430,24 @@ export function logApiResponse(config: Config, event: ApiResponseEvent): void {
     model: event.model,
     type: 'tool',
   });
+
+  // Record OpenTelemetry GenAI semantic convention metrics
+  const { commonAttributes } = recordGenAiClientMetrics(config, event);
+
+  // Record token usage following GenAI conventions
+  if (event.input_token_count > 0) {
+    recordGenAiClientTokenUsage(config, event.input_token_count, {
+      ...commonAttributes,
+      'gen_ai.token.type': GenAiTokenType.INPUT,
+    });
+  }
+
+  if (event.output_token_count > 0) {
+    recordGenAiClientTokenUsage(config, event.output_token_count, {
+      ...commonAttributes,
+      'gen_ai.token.type': GenAiTokenType.OUTPUT,
+    });
+  }
```

Comment:
might be nice to have the lines that emit input/output metrics co-located with the above lines doing the same thing for our 'legacy' metrics

```
  recordTokenUsageMetrics(config, event.input_token_count, {
    model: event.model,
    type: 'input',
  });
  recordTokenUsageMetrics(config, event.output_token_count, {
    model: event.model,
    type: 'output',
  });
```

Reply:
i also wonder if this could go a step further and have a nice little loop over the "types" instead of 5 separate lines of code...

AND i'm wondering, why do the above lines not need to check the input/output token count before emitting the metric, while the new lines do?